### PR TITLE
Ridesharing: Rename Blablacar to Blablalines, price in centime, crofly durations calculated 

### PR DESCRIPTION
**Pull request by kadhikari**
Modifications are the followings:

- Blablacar renamed to Blablalines
- Convertir le prix en centime: Blablacar, klaxit, karos
- Error: ridesharing_journeys absent in Blablalines
- ridesharing_journey[].duration, ridesharing_journey[].durations.walking and ridesharing_journey[].durations.total recalculated.

Concerned tickets: 
- https://jira.kisio.org/browse/NAVITIAII-3212
- https://jira.kisio.org/browse/NAVITIAII-3216
- https://jira.kisio.org/browse/NAVITIAII-3217

### CODE REVIEW

**Review by nonifier — APPROVED** [review]
Nice cleanup ! :pray:

File: source/jormungandr/jormungandr/scenarios/ridesharing/blablalines.py [hunk 1]
```diff
@@ -136,9 +138,13 @@ def _make_response(self, raw_json):
                     type_pb2.GeographicalCoord(lon=res.dropoff_place.lon, lat=res.dropoff_place.lat)
                 )
 
-            res.price = offer.get('price', {}).get('amount')
             currency = offer.get('price', {}).get('currency')
-            res.currency = "centime" if currency == "EUR" else currency
+            if currency == "EUR":
+                res.currency = "centime"
+                res.price = offer.get('price', {}).get('amount') * 100.0
```
Comment (pbench) [inline source/jormungandr/jormungandr/scenarios/ridesharing/blablalines.py:144]: Is `res.price` a float or an int ? 
In other words, is a cast to int needed ?

File: source/jormungandr/jormungandr/scenarios/ridesharing/karos.py [hunk 2]
```diff
@@ -132,7 +134,7 @@ def _make_response(self, raw_json):
                     type_pb2.GeographicalCoord(lon=res.dropoff_place.lon, lat=res.dropoff_place.lat)
                 )
 
-            res.price = offer.get('price', {}).get('amount')
+            res.price = offer.get('price', {}).get('amount') * 100.0
```
Comment (pbench) [inline source/jormungandr/jormungandr/scenarios/ridesharing/karos.py:137]: same as above

File: source/jormungandr/jormungandr/scenarios/ridesharing/klaxit.py [hunk 3]
```diff
@@ -132,7 +134,8 @@ def _make_response(self, raw_json):
                     type_pb2.GeographicalCoord(lon=res.dropoff_place.lon, lat=res.dropoff_place.lat)
                 )
 
-            res.price = offer.get('price', {}).get('amount')
+            # For Klaxit the price is in euro
+            res.price = offer.get('price', {}).get('amount') * 100.0
```
Comment (pbench) [inline source/jormungandr/jormungandr/scenarios/ridesharing/klaxit.py:138]: same as above

File: source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service_manager.py [hunk 4]
```diff
@@ -238,7 +238,16 @@ def build_ridesharing_journeys(self, from_pt_obj, to_pt_obj, period_extremity):
             start_teleport_section.origin.CopyFrom(from_pt_obj)
             start_teleport_section.destination.CopyFrom(pb_rsj_pickup)
             start_teleport_section.length = int(crowfly_distance_between(from_coord, pickup_coord))
-            start_teleport_section.duration = 0
+
+            # We take departure to pickup duration
+            start_teleport_section.duration = (
+                int(rsj.origin_pickup_duration)
+                if getattr(rsj, 'origin_pickup_duration', None) is not None
+                else 0
```
Comment (pbench) [inline source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service_manager.py]: Maybe simpler like this ? :
```suggestion
                int(getattr(rsj, 'origin_pickup_duration', 0))
```

File: source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service_manager.py [hunk 5]
```diff
@@ -307,7 +315,15 @@ def build_ridesharing_journeys(self, from_pt_obj, to_pt_obj, period_extremity):
             end_teleport_section.origin.CopyFrom(pb_rsj_dropoff)
             end_teleport_section.destination.CopyFrom(to_pt_obj)
             end_teleport_section.length = int(crowfly_distance_between(dropoff_coord, to_coord))
-            end_teleport_section.duration = 0
+
+            # We take dropoff to destination duration
+            end_teleport_section.duration = (
+                int(rsj.dropoff_dest_duration) if getattr(rsj, 'dropoff_dest_duration', None) is not None else 0
```
Comment (pbench) [inline source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service_manager.py]:
```suggestion
            int(getattr(rsj, 'dropoff_dest_duration', 0))
```

File: source/jormungandr/jormungandr/scenarios/ridesharing/blablalines.py (hunk 1 above)
Comment (kadhikari) [reply source/jormungandr/jormungandr/scenarios/ridesharing/blablalines.py:144]: Int or float, once multiplied with a float will change it to float. The price is in float in navitia.

File: source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service_manager.py (hunk 5 above)
Comment (kadhikari) [reply source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service_manager.py]: :+1:

File: source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service_manager.py (hunk 4 above)
Comment (kadhikari) [reply source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service_manager.py]: :+1:

File: source/jormungandr/jormungandr/scenarios/ridesharing/blablalines.py (hunk 1 above)
Comment (pbench) [reply source/jormungandr/jormungandr/scenarios/ridesharing/blablalines.py:144]: Allright, everything's good then :+1:

**Review by benoit-bst — APPROVED** [review]
nice shot, dude !

**Review by azime — APPROVED** [review]

**Pull request merged**